src/bonds.cpp

```diff
@@ -174,6 +189,25 @@ void FENEWCABond::setEnergyFunction(const ParticleVector& particles) {
         return -half_force_constant * max_distance_squared * std::log(1.0 - squared_distance / max_distance_squared) +
                wca;
     };
+    forceFunc = [&](Geometry::DistanceFunction distance) -> std::vector<IndexAndForce> {
+        const Point ba = distance(particles[indices[0]].pos, particles[indices[1]].pos); // b->a
+        const auto squared_distance = ba.squaredNorm();
+        if (squared_distance >= max_distance_squared) {
+            throw std::runtime_error("Fene+WCA potential: Force undefined for distances greater than rmax.");
+        }
+        double wca_force = 0.0;
+        constexpr auto two_to_the_power_of_two_sixths = 1.01944064370214482816981563263103378007648819; // 2^((1/6)^2)
+        if (squared_distance <= sigma_squared * two_to_the_power_of_two_sixths) {
+            double sigma6 = sigma_squared / squared_distance;
+            sigma6 = sigma6 * sigma6 * sigma6;
+            wca_force = -24.0 * epsilon * (2.0 * sigma6 * sigma6 - sigma6) / ba.norm();
+        }
+        const auto magnitude =
+            -(2.0 * half_force_constant * ba.norm() / (1.0 - squared_distance/max_distance_squared) + wca_force);
```

Comment:
save `ba.norm()` in a variable and use it both above and below. This would save two calls to sqrt.

Reply:
Updated to
```c++
    forceFunc = [&](Geometry::DistanceFunction calculateDistance) -> std::vector<IndexAndForce> {
        const Point ba = calculateDistance(particles[indices[0]].pos, particles[indices[1]].pos); // b->a
        const auto distance = ba.norm();
        const auto squared_distance = distance * distance;
        if (squared_distance >= max_distance_squared) {
            throw std::runtime_error("Fene+WCA potential: Force undefined for distances greater than rmax.");
        }
        double wca_force = 0.0;
        constexpr auto two_to_the_power_of_two_sixths = 1.01944064370214482816981563263103378007648819; // 2^((1/6)^2)
        if (squared_distance <= sigma_squared * two_to_the_power_of_two_sixths) {
            double sigma6 = sigma_squared / squared_distance;
            sigma6 = sigma6 * sigma6 * sigma6;
            wca_force = -24.0 * epsilon * (2.0 * sigma6 * sigma6 - sigma6) / distance;
        }
        const auto magnitude =
            -(2.0 * half_force_constant * distance / (1.0 - squared_distance/max_distance_squared) + wca_force);
```
Fixed with f1feccc.